conda/core/solve.py

```diff
@@ -450,7 +455,27 @@ def solve_for_transaction(self, deps_modifier=NULL, prune=NULL, ignore_pinned=NU
                                   self.specs_to_remove, self.specs_to_add)
                 # TODO: Only explicitly requested remove and update specs are being included in
                 #   History right now. Do we need to include other categories from the solve?
-                return UnlinkLinkTransaction(stp)
+
+        conda_newer_spec = MatchSpec('conda >%s' % CONDA_VERSION)
+        if not any(conda_newer_spec.match(prec) for prec in link_precs):
+            conda_newer_records = sorted(query_all(self.channels, self.subdirs, conda_newer_spec),
+                                         key=lambda x: VersionOrder(x.version))
+            if conda_newer_records:
+                latest_version = conda_newer_records[-1].version
+                sys.stderr.write(dedent("""
+
+                ==> WARNING: A newer version of conda exists. <==
```

Comment:
Can this instead be:
```
==> INFO: A newer version of conda is available!  <==
```

Reply:
Warning is could be considered something in negative sense.

Reply:
I think it is negative.  I think we WANT people to actively be upgrading, and feel "uncomfortable" if they're on a non-latest version.  Pretty tired of people reporting errors that have already been fixed :(

Reply:
Will this warning be thrown on each operation involving solver even if I have pinned conda?

Reply:
Yes, it will.  pip has basically the exact same behavior.  Are you concerned with being too annoying?

Reply:
I really don't mind being annoying like this when people are using outdated versions of conda.  Lots of software does it.

In fact, I'd rather *just* be annoying, and change the default setting for `auto_update_conda` to `false`.  Crystal didn't want to do that though.

Reply:
In the scenario when the base environment is controlled by an admin and user is not allowed to modify it, I think it can be annoying. I think this warrants a discussion with a wider audience. Personally, I am okay with it being displayed always, but I'd still vote for the log level to be INFO instead of warning. Pip also allows users to control this behaviour with:
```
[global]
disable-pip-version-check = True
```
Maybe we could have something similar for Conda too?